navigator/backend/src/main/scala/com/digitalasset/navigator/graphql/GraphQLSchema.scala

```diff
@@ -317,7 +317,8 @@ final class GraphQLSchema(customEndpoints: Set[CustomEndpoint[_]]) {
           ListType(ExercisedEventType),
           resolve =
             context => context.ctx.ledger.exercisedEventsOf(context.value, context.ctx.templates)),
-        Field("argument", JsonType.ApiRecordType, resolve = _.value.argument)
+        Field("argument", JsonType.ApiRecordType, resolve = _.value.argument),
+        Field("agreementText", OptionType(StringType), resolve = _.value.agreementText)
```

Comment:
This does not conform to the schema (should be caught by a test).

In `schema.graphql`, you have `agreementText: String!`, here you use `OptionType(StringType)`.
`String!` means a non-optional string in GraphQL, `OptionType` means the backend may produce missing values for the `agreementText` property.

Use `agreementText: String` in the schema to specify that the property is optional.

Reply:
Done. Thanks.

Reply:
Speaking of which: is this the equivalent of the graphql schema in the world of queries? https://github.com/digital-asset/daml/blob/d6236a9f1ddc6fc9e8851ffd95830dabdba7d6eb/navigator/frontend/src/api/Queries.ts#L38

Also, the spray.json renderer for `Option[_]` uses `JsNull` for the `None` case. Is this properly supported by GraphQl (and the respective libraries we're using)?

Reply:
The `Queries.ts` file is indeed the TypeScript definition for the shape of our GraphQL API responses.

It is autogenerated from `schema.graphql`. Or rather, it should be - if you edited it by hand, please run `make update-graphql-types` in `navigator/frontend` instead. The reason why this is not done as part of the Bazel build is that I was too lazy to configure the frontend project to look for source files outside of the source tree (where Bazel generated files are stored).

Reply:
oh, that yeah, I definitely edited it by hand. I'll run the make command.

Reply:
https://github.com/digital-asset/daml/issues/1160